spring-vault-core/src/main/java/org/springframework/vault/support/VaultMetadataResponseWrapper.java

```diff
@@ -0,0 +1,17 @@
+package org.springframework.vault.support;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class VaultMetadataResponseWrapper {
```

Comment:
With custom deserialization, there's no need for this type. An inner class (`VaultMetadataResponseWrapper 
 extends VaultResponseSupport<Map<String, Object>>`) in `VaultKeyValueMetadataTemplate` should be sufficient.